src/Nethermind/Nethermind.Trie/Pruning/CommitSetQueue.cs

```diff
@@ -0,0 +1,116 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Nethermind.Core.Collections;
+using Nethermind.Core.Crypto;
+
+namespace Nethermind.Trie.Pruning;
+
+public class CommitSetQueue
+{
+    private SortedSet<BlockCommitSet> _queue = new();
+
+    public int Count
+    {
+        get
+        {
+            lock (_queue)
```

Comment:
Does the commit set queue even has multi-threaded access?
We have block processing queue so the write access should be single threaded?
Alternatively could use `ReaderWriterLockSlim` for multiple readers.

Reply:
It can write and prune at the same time.